corehq/apps/userreports/models.py

```diff
@@ -778,7 +785,53 @@ def ui_filters(self):
     @property
     @memoized
     def charts(self):
-        return [ChartFactory.from_spec(g._obj) for g in self.configured_charts]
+        if (
+            self.config_id and self.configured_charts
+            and toggles.SUPPORT_EXPANDED_COLUMN_IN_REPORTS.enabled(self.domain)
+        ):
+            configured_charts = deepcopy(self.configured_charts)
+            for chart in configured_charts:
+                if chart['type'] == 'multibar':
+                    chart['y_axis_columns'] = self._get_expanded_y_axis_cols_for_multibar(chart['y_axis_columns'])
+            return [ChartFactory.from_spec(g._obj) for g in configured_charts]
+        else:
+            return [ChartFactory.from_spec(g._obj) for g in self.configured_charts]
+
+    def _get_expanded_y_axis_cols_for_multibar(self, original_y_axis_columns):
+        y_axis_columns = []
+        try:
+            for y_axis_column in original_y_axis_columns:
+                if isinstance(y_axis_column, dict):
+                    column_id = y_axis_column['column_id']
+                else:
+                    column_id = y_axis_column
+                column_config = self.report_columns_by_column_id[column_id]
+                if column_config.type == 'expanded':
+                    expanded_columns = self.get_expanded_columns(column_config)
+                    for column in expanded_columns:
+                        y_axis_columns.append({
+                            'column_id': column.slug,
+                            'display': column.header
+                        })
+                else:
+                    y_axis_columns.append(y_axis_column)
+        except DataSourceConfigurationNotFoundError:
```

Comment:
Can you explain more about what this catches? If the data source does not exist, how do we show a report?

Reply:
It catches the situation where the data source is yet to be created and the report is being validated or loaded.

This was definitely happening and causing error. I believe when we review a report and then eventually save to view it, those are both different data sources, so if there is any delay in that, the report will throw error if its validated or in this case even loaded and charts are requested. Happy to confirm on when exactly this happens.

Reply:
Added comment about this to be explicit in code 
https://github.com/dimagi/commcare-hq/pull/30873/commits/9b6bd74d8167daffcc4b2c2b2e659a00cbb774ee